src/Sleep.js

```diff
@@ -1,38 +1,25 @@
-class Sleep {
+import StatsParent from '../src/StatsParent'
+class Sleep extends StatsParent{
   constructor(sleepData, userID) {
+    super();
     this.sleepData = sleepData;
     this.userID = userID;
-  }
-
-  findUser() {
-    return this.sleepData.filter(user => user.userID === this.userID);
-  }
-
-  returnWeekOfData(week, userData) {
-    return [...userData].splice((-7 * week), 7);
-  }
-
-  returnWeek(week) {
-    var specificUser = this.findUser()
-    return [...specificUser].splice(-7 * week, 7).map(day => day.date);
+    this.finder = this.findUser(this.sleepData,this.userID)
```

Comment:
This is interesting, can you tell me why this is superior to just calling findUser() and do you think I should do it with Activity? @MattTuring

Reply:
Its less repetitive code for the sake of DRY programing. It would be best to do this to statsParent and call it in the extended classes.

Reply:
Sounds good, I made an issue!